pennylane/hf/integrals.py

```diff
@@ -308,6 +308,207 @@ def overlap_integral(*args):
+        rc (array[float]): distance between the center of the Hermite Gaussian and the origin
+
+    Returns:
+        array[float]: expansion coefficients for each Gaussian combination
+
+    **Example**
+
+    >>> alpha = np.array([3.42525091])
+    >>> beta =  np.array([3.42525091])
+    >>> t = 0
+    >>> e = 1
+    >>> rc = 1.5
+    >>> _hermite_moment(alpha, beta, t, e, rc)
+    array([1.0157925])
+    """
+    p = anp.array(alpha + beta)
+
+    if t > e:
+        return 0.0
+    if e == 0 and t != 0:
+        return 0.0
+    if e == 0 and t == 0:
+        return anp.sqrt(anp.pi / p)
+    m = (
+        _hermite_moment(alpha, beta, t - 1, e - 1, rc) * t
+        + _hermite_moment(alpha, beta, t, e - 1, rc) * rc
+        + _hermite_moment(alpha, beta, t + 1, e - 1, rc) / (2 * p)
+    )
+    return m
+
+
+def gaussian_moment(la, lb, ra, rb, alpha, beta, e, rc):
+    r"""Compute one-dimensional multipole moment integral for two primitive Gaussian functions.
+
+    The multipole moment integral in one dimension is defined as
+
+    .. math::
+
+        S_{ij}^e = \left \langle G_i | q_C^e | G_j \right \rangle,
+
+    where :math:`G` is a Gaussian function at dimension :math:`q = x, y, z` of the Cartesian
+    coordinates system, :math:`e` is the multipole moment order and :math:`C` is the origin of the
+    Cartesian coordinates. The integrals can be evauated as
+    [`Helgaker (1995) p803 <https://www.worldscientific.com/doi/abs/10.1142/9789812832115_0001>`_]
+
+    .. math::
+
+        S_{ij}^e = \sum_{t=0}^{\mathrm{min}(i+j, \ e)} E_t^{ij} M_t^e,
+
+    where :math:`E` and :math:`M` are the Hermite Gaussian expansion coefficient and the Hermite
+    moment integral, respectively, that can be computed recursively.
+
+    Args:
+        la (integer): angular momentum for the first Gaussian function
+        lb (integer): angular momentum for the second Gaussian function
+        ra (float): position of the first Gaussian function
+        rb (float): position of the second Gaussian function
+        alpha (array[float]): exponent of the first Gaussian function
+        beta (array[float]): exponent of the second Gaussian function
+        e (integer): order of the multipole moment
+        rc (array[float]): distance between the center of the Hermite Gaussian and the origin
+
+    Returns:
+        array[float]: one-dimensional multipole moment integral between primitive Gaussian functions
+
+    **Example**
+
+    >>> la, lb = 0, 0
+    >>> ra, rb = np.array([2.0]), np.array([2.0])
+    >>> alpha = np.array([3.42525091])
+    >>> beta = np.array([3.42525091])
+    >>> e = 1
+    >>> rc = 1.5
+    >>> gaussian_moment(la, lb, ra, rb, alpha, beta, e, rc)
+    array([1.0157925])
+    """
+    s = 0.0
+    for t in range(min(la + lb + 1, e + 1)):
+        s = s + expansion(la, lb, ra, rb, alpha, beta, t) * _hermite_moment(alpha, beta, t, e, rc)
+
+    return s
+
+
+def generate_moment(basis_a, basis_b, e, idx):
```

Comment:
This can be a good trial for renaming functions more widely. Calling this `moment_integral` makes sense to me, which means that we should rename the function inside. Could we use `_moment_integral`? Or maybe here use a longer name, `compute_moment_integral`? `calculate_moment_integral`?